### DISCUSSION TITLE
project_ref exposed to Postgres / some default vault items set

### DISCUSSION THREAD

**Discussion by gc-ft** [supabase/supabase#29036] [Feature Requests]
Hi,

especially with the new branching feature, it would be extremely useful to be able to retrieve the project_ref through a Postgres call, so most likely reading it from a setting variable that Supabase creates when Postgres launches on the project.

This would allow building seeds which use URLs for webhooks and the like that point to the right project.

A second thing that would be very useful is if Supabase would publish some of its API variables in the Postgres Vault, at least the service_role key, but in theory also the project_ref here instead of in the Postgres settings if that is easier to achieve.

Any chance that this will be implemented? Right now automated CI/CD branches for testing is hard if not impossible to implement because each time one would have to change some settings once the branch is created since only then the project_ref and keys are known. If they would be published in postgres vault then we could get them from there in the CI/CD pipeline or when seeding the branch db.

Fabian

**Comment by sweatybridge** [comment]
I'm closing this as a duplicate of https://github.com/orgs/supabase/discussions/12813
